.github/workflows/bot-pr-updated.yml

```diff
@@ -22,3 +22,37 @@ jobs:
         with:
           name: patch
           path: format.patch
+  abidiff:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.author_association == 'COLLABORATOR' || github.event.pull_request.author_association == 'MEMBER' || github.event.pull_request.author_association == 'OWNER'
+    env:
+      CMAKE_FLAGS: -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=DEBUG -DGINKGO_BUILD_TESTS=OFF -DGINKGO_BUILD_EXAMPLES=OFF -DGINKGO_BUILD_BENCHMARKS=OFF -DGINKGO_BUILD_HWLOC=OFF -DGINKGO_BUILD_REFERENCE=OFF -DGINKGO_BUILD_OMP=OFF -DGINKGO_BUILD_CUDA=OFF -DGINKGO_BUILD_DPCPP=OFF
+    steps:
+    - name: Checkout the new code (shallow clone)
+      uses: actions/checkout@v2
+      with:
+        path: new
+        ref: ${{ github.event.pull_request.head.ref }}
+    - name: Checkout the old code (shallow clone)
+      uses: actions/checkout@v2
+      with:
+        path: old
+        ref: develop
```

Comment:
This might produce clutter when creating a PR to merge to master. I think it's best to filter by PR target to `develop`? Otherwise if possible use the PR target instead here.

Reply:
Maybe it is better also to always use master as the old ref ? Maybe it is more important that we are ABI compatible with the current release tag (on master) rather than the target branch ? 

We could then also additionally cache the master build in a docker container and speed up this job by two times. Maybe this is not that important as we dont really build any tests or backend kernels.

Reply:
I guess you mean `${{ github.event.pull_request.base.ref }}`?

Reply:
master is a good point because we need to keep public interface during the minor version update
However, I also wonder there are too many changes such that we can not get good information from the current branch

Reply:
I don't think we should make the master the base, for every PR, otherwise, with every new PR we will get more and more lines of `ABIDIFF` change. If this was enforcing ABI stability (which is hard, IHMO), then master would be a good base point, indeed.